js/main.js

```diff
@@ -0,0 +1,166 @@
+"use strict";
+
+var gameSettings = {
+  width: 0,
+  height: 0,
+  bombsAmount: 0,
+  cellsArray: [],
+  cellsLeft: 0
+};
+
+function initGame(fieldDimension, bombsAmount){
+  gameSettings.height = fieldDimension;
+  gameSettings.width = fieldDimension;
+  gameSettings.bombsAmount = bombsAmount;
+  console.log(gameSettings);
+  startGame(gameSettings);
+}
+
+function startGame(gameSettings){
+  var cellsContainer = document.getElementsByClassName('container')[0];
+  hideElement('mainMenu');
+  gameSettings.cellsArray = generatePlayField(gameSettings, cellsContainer);
+  gameSettings.cellsLeft = gameSettings.height * gameSettings.width;
+  showElement('gameMenu');
+}
+
+function cancelGame(){
+  resetContainer();
+  hideElement('gameMenu');
+  showElement('mainMenu');
+}
+
+function restartGame(){
+  resetContainer();
+  startGame(gameSettings);
+}
+
+function leftClickHandler(element, row, column){
+  if(hasFlag(element)){
+    updateBombsLeft(1);
+  }
+  if(hasMine(element)){
+    endGame();
+  }
+  
+  showCellValue(element);
+  
+  if(!element.hasAttribute('value')){
+    openZone(element, row, column);
+  }
+}
+
+function rightClickHandler(element, event){
+  preventContextMenu(event);
+  if(!element.hasAttribute('clicked')){
+    swapFlag(element);
+  }
+}
+
+function swapFlag(element){
+  if(element.innerHTML === 'X'){
+    element.innerHTML = '';
+    element.removeAttribute('flag');
+    updateBombsLeft(1);
+  }else{
+    element.innerHTML = 'X';
+    element.setAttribute('flag', '');
```

Comment:
For own attributes better use attribute 'data-', which is used to store additional information about the item.